x-pack/plugins/security_solution/public/detections/components/endpoint_responder/use_responder_action_item.tsx

```diff
@@ -39,10 +41,18 @@ export const useResponderActionItem = (
     [eventDetailsData]
   );
 
+  const { agentStatus } = useHostIsolationStatus({ agentId: endpointId });
+
   return useMemo(() => {
     const actions: JSX.Element[] = [];
 
-    if (isResponseActionsConsoleEnabled && !isAuthzLoading && canAccessResponseConsole && isAlert) {
+    if (
+      isResponseActionsConsoleEnabled &&
+      !isAuthzLoading &&
+      agentStatus !== HostStatus.UNENROLLED &&
```

Comment:
So this changes the behaviour of the menu where the items on it appear and disapear depending on what alert you are looking at. That behaviour is frustrating and confusing because the user will not knw why a certain menu item is not present on this specific  alert while it appears on another. The only case where the menu item should not be there is when the feature is not authorized or this is not an alert. All other cases, the menu item shouild be there, but possibly disabled with a tooltip explaining why.

Also - the `<ResponderContextMenuItem />` component already checks for un-enrolled agents and it disables the menu item and adds to it a tooltip ([see here](https://github.com/elastic/kibana/blob/4227348b9579900e5142a63720bb108c2c33f48a/x-pack/plugins/security_solution/public/detections/components/endpoint_responder/responder_context_menu_item.tsx#L65-L73))